### PR TITLE
feat: Statsing provider

### PR DESCRIPTION
**Pull request by jenshenneberg**
## This PR
- Adding a minimal Statsig provider integration

### Notes
The implementation is minimal - only `ResolveBooleanValue` is implemented.

### How to test
Create a free Statsig account and use as described in README.md

### CODE REVIEW

File: src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs
Comment (beeme1mr) [inline src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: I'm checking on this to see if we can release it soon.

File: src/OpenFeature.Contrib.Providers.Statsig/StatsigProvider.cs
Comment (beeme1mr) [inline src/OpenFeature.Contrib.Providers.Statsig/StatsigProvider.cs:54]: Will the Statsig SDK throw if you pass a default value of `true`? The OpenFeature SDK will catch any exception thrown and return the default value. It may not be necessary to explicitly check and throw here.

**Review by beeme1mr — APPROVED** [review]
Overall, it looks good. I left a few comments but I think we can merge this when you're ready.

File: src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs
Comment (jenshenneberg) [reply src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: Thank you, if it's close we might as well wait for that.

File: src/OpenFeature.Contrib.Providers.Statsig/StatsigProvider.cs
Comment (jenshenneberg) [reply src/OpenFeature.Contrib.Providers.Statsig/StatsigProvider.cs:54]: The Statsig SDK does not have a fallback option for boolean evaluation, I've created a ticket for it [here](https://github.com/statsig-io/dotnet-sdk/issues/33). 
I felt it would be dangerous to allow passing in a default value of `true` as the SDK will always return false for non defined feature gates.

**Comment by jenshenneberg** [comment]
Thanks for having a look. 😄   I'll do some more testing and wait for a decision on a new release of https://github.com/open-feature/dotnet-sdk/ before merging

File: src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs
Comment (beeme1mr) [reply src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: We released 1.5 last night.

Comment (jenshenneberg) [reply src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: Thanks, @beeme1mr. Would you be ok with me doing the update [here](https://github.com/open-feature/dotnet-sdk-contrib/blob/533dfa652011b645d48c5d17dbdfe39b64a8eadf/build/Common.props#L27) as part of this PR, or does it need to be in a separate one?

Comment (beeme1mr) [reply src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: Sure, that would be fine with me.

Comment (jenshenneberg) [reply src/OpenFeature.Contrib.Providers.Statsig/EvaluationContextExtensions.cs:26]: In the end I've decided to include it in this PR as it was a minor update. Please let me know if you would like me to change that so it goes in its own PR.

**Comment by jenshenneberg** [comment]
Updated to `OpenFeature` 1.5.* as part of this PR. Please merge this PR if you feel it's ready. Thanks for your input and help with this. 😃

**Pull request merged**
